Digitall.Persistance.EF/UnitOfWork.cs

```diff
@@ -0,0 +1,35 @@
+﻿using Digitall.Warehouse.Application.Abstractions.Persistence;
+using Digitall.Warehouse.Domain.Abstraction;
+using Microsoft.EntityFrameworkCore;
+
+namespace Digitall.Persistance.EF
+{
+    public class UnitOfWork(WarehouseDbContext dbContext) : IUnitOfWork
+    {
+        private readonly WarehouseDbContext _dbContext = dbContext;
+
```

Comment:
You should inject CategoryRepository and all other repositories here. Then you can use unitofwork only instead of injecting each separate repository where you need it.

Reply:
You did say that already - I've understood :)